R/create_query_data.R

```diff
@@ -0,0 +1,894 @@
+  values <- unclass(obj)
+  name <- values$name
+  assert_character_scalar(name,
+                          optional = TRUE)
+  id <- values$id
+  assert_integer_scalar(id,
+                        optional = TRUE)
+  scope <- values$scope
+  assert_character_scalar(scope,
+                          values = c("BROAD", "NARROW"))
+
+  if (is.null(values$id) && is.null(values$name)) {
+    abort("Either id or name has to be non null.")
+  }
+  if (!is.null(values$id) && !is.null(values$name)) {
+    abort("Either id or name has to be null.")
+  }
+  obj
+}
+
+#' Returns a Character Representation of a `smq_select()` Object
+#'
+#' The function returns a character representation of a `smq_select()` object.
+#' It can be used for error messages for example.
+#'
+#' @param x A `smq_select()` object
+#'
+#' @param ... Not used
+#'
+#' @return A character representation of the `smq_select()` object
+#'
+#' @author Stefan Bundfuss
+#'
+#' @seealso [smq_select()]
+#'
+#' @keywords dev_utility
+#'
+#' @export
+#'
+#' @examples
+#'
+#' format(smq_select(id = 42, scope = "NARROW"))
+format.smq_select <- function(x, ...) {
+  paste0("smq_select(name = ",
+         dquote(x$name),
+         ", id = ",
+         format(x$id),
+         ", scope = ",
+         dquote(x$scope),
+         ")")
+}
+
+#' Create an `sdg_select` object
+#'
+#' @param name Name of the query used to select the definition of the query
+#'   from the central database.
+#'
+#'   The query with `GROUP_NAME == "<specified value>"` is selected.
+#'
+#' @param id Identifier of the query used to select the definition of the query
+#'   from the central database.
+#'
+#'   The query with `GROUP_ID == <specified value>` is selected.
+#'
+#' @details Exactly one `name` or `id` must be specified.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @seealso [create_query_data()], [query()]
+#'
+#' @keywords source_specifications
+#'
+#' @export
+#'
+#' @return An object of class "sdg_select".
+sdg_select <- function(name=NULL,
+                       id = NULL) {
+  out <- list(
+    name = name,
+    id = id
+  )
+  class(out) <- c("sdg_select", "list")
+  validate_sdg_select(out)
+}
+
+#' Validate an object is indeed a `sdg_select` object
+#'
+#' @param obj An object to be validated.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @seealso [sdg_select()]
+#'
+#' @export
+#'
+#' @return The original object.
+validate_sdg_select <- function(obj) {
+  assert_that(inherits(obj, "sdg_select"))
+  values <- unclass(obj)
+  name <- values$name
+  assert_character_scalar(name,
+                          optional = TRUE)
+  id <- values$id
+  assert_integer_scalar(id,
+                        optional = TRUE)
+  if (is.null(values$id) && is.null(values$name)) {
+    abort("Either id or name has to be non null.")
+  }
+  if (!is.null(values$id) && !is.null(values$name)) {
+    abort("Either id or name has to be null.")
+  }
+  obj
+}
+
+#' Returns a Character Representation of a `sdg_select()` Object
```

Comment:
same as for `format.smq_select` -  this seems not used.

Reply:
See my comment there.